frontend/www/js/omegaup/problem/edit.ts

```diff
@@ -1,46 +1,239 @@
+              usernameOrEmail: username,
+            })
+              .then(() => {
+                ui.success(T.adminRemoved);
+                this.refreshProblemAdmins();
+              })
+              .catch(ui.apiError);
+          },
+          'add-group-admin': (groupAlias: string): void => {
+            api.Problem.addGroupAdmin({
+              problem_alias: payload.alias,
+              group: groupAlias,
+            })
+              .then(() => {
+                ui.success(T.groupAdminAdded);
+                this.refreshProblemAdmins();
+              })
+              .catch(ui.apiError);
+          },
+          'remove-group-admin': (groupAlias: string): void => {
+            api.Problem.removeGroupAdmin({
+              problem_alias: payload.alias,
+              group: groupAlias,
+            })
+              .then(() => {
+                ui.success(T.groupAdminRemoved);
+                this.refreshProblemAdmins();
+              })
+              .catch(ui.apiError);
+          },
+          'select-version': (
+            selectedRevision: types.ProblemVersion,
+            updatePublished: boolean,
+          ) => {
+            api.Problem.selectVersion({
+              problem_alias: payload.alias,
+              commit: selectedRevision.commit,
+              update_published: updatePublished,
+            })
+              .then(response => {
+                problemEdit.publishedRevision = selectedRevision;
+                ui.success(T.problemVersionUpdated);
+              })
+              .catch(ui.apiError);
+          },
+          'runs-diff': (
+            versions: types.CommitRunsDiff,
+            selectedCommit: omegaup.Commit,
+          ) => {
+            api.Problem.runsDiff({
+              problem_alias: payload.alias,
+              version: selectedCommit.version,
+            })
+              .then(response => {
+                Vue.set(
+                  versions.runsDiff,
+                  selectedCommit.version,
+                  response.diff,
                 );
               })
               .catch(ui.apiError);
           },
+          remove: (problemAlias: string) => {
+            api.Problem.delete({ problem_alias: problemAlias })
+              .then(response => {
+                window.location.href = '/problem/mine/';
+              })
+              .catch(ui.apiError);
+          },
         },
       });
     },
+    methods: {
+      refreshProblemAdmins: (): void => {
+        api.Problem.admins({ problem_alias: payload.alias })
+          .then(data => {
+            problemEdit.initialAdmins = data.admins;
+            problemEdit.initialGroups = data.group_admins;
+          })
+          .catch(ui.apiError);
+      },
+    },
+    data: {
+      initialAdmins: payload.admins,
+      initialGroups: payload.groupAdmins,
+      log: <types.ProblemVersion[]>payload.log,
+      publishedRevision: <types.ProblemVersion>payload.publishedRevision,
```

Comment:
como esto ya no es una literal de arreglo / objeto, no es necesario agregar la anotación de tipo:

```suggestion
      publishedRevision: payload.publishedRevision,
```

Reply:
Listo